api/python/provisioner/commands/validator/validator.py

```diff
@@ -348,22 +347,11 @@ def validate(self, path: Path):
         """
         super().validate(path)
 
-        if self.hash_type.value not in hashlib.algorithms_available:
-            raise ValidationError(f"Hash type '{self.hash_type.value}' is not"
-                                  "supported by Python's `hashlib` module.")
+        hash_obj = utils.calc_hash(path, self.hash_type)
 
-        hash_method = getattr(hashlib, self.hash_type.value)()
-
-        with open(path, 'rb') as fh:
-            while True:
-                data = fh.read(4096)
-                if not data:
-                    break
-                hash_method.update(data)
-
-        if not compare_digest(hash_method.digest(), self.hash_sum):
+        if not compare_digest(hash_obj.digest(), self.hash_sum):
```

Comment:
Please add a comment here mentioning what this digest is and how we are using it in this context of code.

Reply:
done